In [1]:
import pandas as pd
from pathlib import Path

dataset_path = "datasets/Titanic-Dataset.csv"

df = pd.read_csv(dataset_path)


In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


In [3]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [4]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='str')

In [5]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [6]:
df['Age'] = df['Age'].fillna(df['Age'].mean())

Q.1 How many Male and Female were there?
There were 577 males and 314 females.

In [7]:
total_count = df["Sex"].value_counts()
print(total_count)

Sex
male      577
female    314
Name: count, dtype: int64


Q2. How many percentage of Passengers Survived?
Approximately 38.38% of passengers survived.

In [8]:
ppl_sur = df["Survived"].value_counts(normalize=True)
print(ppl_sur)

Survived
0    0.616162
1    0.383838
Name: proportion, dtype: float64


Q3. How many males and how many females survived?
74.20% of females survived, and 18.89% of males survived.

In [9]:
gender_sur = df.groupby('Sex')['Survived'].mean()
print(gender_sur)

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64


Q4. What is survival rate by passenger class?
Survival rates by passenger class:
- Class 1: 62.96%
- Class 2: 47.28%
- Class 3: 24.24%

In [10]:
class_survival = df.groupby("Pclass")["Survived"].mean()

print(class_survival)


Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


Q5. How many were children and how many survived?
There were 113 children (under 18) on board, and 61 of them survived.

In [11]:
children = df[(df["Age"] < 18)]

print("Total children :", len(children))

survived_children = children[children["Survived"] == 1]

print("Survived Children :", len(survived_children))



Total children : 113
Survived Children : 61


Q6. Who paid the highest fare?
The passengers who paid the highest fare are:
- Cardeza, Mr. Thomas Drake Martinez: 512.33
- Ward, Miss. Anna: 512.33
- Lesurer, Mr. Gustave J: 512.33

In [12]:
highest_fare = (
    df[["Name", "Fare"]]
    .sort_values(by="Fare", ascending=False)
)

print(highest_fare.head(10))

                                                Name      Fare
679               Cardeza, Mr. Thomas Drake Martinez  512.3292
258                                 Ward, Miss. Anna  512.3292
737                           Lesurer, Mr. Gustave J  512.3292
88                        Fortune, Miss. Mabel Helen  263.0000
438                                Fortune, Mr. Mark  263.0000
341                   Fortune, Miss. Alice Elizabeth  263.0000
27                    Fortune, Mr. Charles Alexander  263.0000
742            Ryerson, Miss. Susan Parker "Suzette"  262.3750
311                       Ryerson, Miss. Emily Borie  262.3750
299  Baxter, Mrs. James (Helene DeLaudeniere Chaput)  247.5208


Q.7 Where Passengers Embarked from and which had maximum passengers?
Passengers embarked from:
- Cherbourg (C): 168 passengers
- Queenstown (Q): 77 passengers
- Southampton (S): 644 passengers

The maximum passengers embarked from Southampton (S).

In [13]:
embarked_class = df.groupby("Embarked")["Embarked"].count()
print("Embarked :", embarked_class)
max_embarked = embarked_class.idxmax()
print(max_embarked)


Embarked : Embarked
C    168
Q     77
S    644
Name: Embarked, dtype: int64
S


Q.8 Ratio of how many survived from each embark
Survival rates by embarkation point:
- Cherbourg (C): 55.36%
- Queenstown (Q): 38.96%
- Southampton (S): 33.70%

In [14]:
embarked_survival = df.groupby('Embarked')['Survived'].mean()
print(embarked_survival)

Embarked
C    0.553571
Q    0.389610
S    0.336957
Name: Survived, dtype: float64
